lib/services/test-execution-service.ts

```diff
@@ -122,8 +145,36 @@ class TestExecutionService implements ITestExecutionService {
 			if (this.$options.debugBrk) {
 				karmaConfig.browserNoActivityTimeout = 1000000000;
 			}
+
+			karmaConfig.projectDir = this.$projectData.projectDir;
 			this.$logger.debug(JSON.stringify(karmaConfig, null, 4));
-			new KarmaServer(karmaConfig).start();
+
+			let karmaRunner = require("child_process").fork(path.join(__dirname, "karma-execution.js"));
+			karmaRunner.send({karmaConfig: karmaConfig});
+			karmaRunner.on("message", (karmaData: any) => {
+				fiberBootstrap.run(() => {
+					this.$logger.trace("## Unit-testing: Parent process received message", karmaData);
+					let port: string;
+					if(karmaData.url) {
+						port = karmaData.url.port;
```

Comment:
I need the port for configuration below (next `if`). However, I've replaced the hard-coded  `localhost:${port}` with `host` which contains both `hostname` and `port` (check next line)